code/modules/atmospherics/gasmixtures/reactions.dm

```diff
@@ -448,7 +448,10 @@
 	air.adjust_moles(GAS_PLASMA, -2*reaction_efficency)
 
 	//clamps by a minimum amount in the event of an underflow.
-	SSresearch.science_tech.add_point_type(TECHWEB_POINT_TYPE_DEFAULT, clamp((reaction_efficency**2)*BZ_RESEARCH_AMOUNT,0.01,BZ_RESEARCH_MAX_AMOUNT))
+	if(isturf(holder))
```

Comment:
This doesn't account for reactions happening in tanks or canisters

Reply:
that's the point?

Reply:
Canisters and tanks exist on the station

Reply:
the point is still being able to make research points via turfs, tanks, canisters and not with math inside a computer. You are making it uses turfs only

Reply:
yes, i want it to be turf only, but you didn't want it so whatever